# Condition and Conditional Statements

Memodifikasi cara kerja fungsi, tergantung pada masukannya.

## Conditions (Kondisi)

Dalam pemrograman, kondisi adalah pernyataan yang bernilai Benar atau Salah. Ada banyak cara berbeda untuk menulis kondisi dalam Python, tetapi beberapa cara paling umum untuk menulis kondisi hanya membandingkan dua nilai yang berbeda. Misalnya, Anda dapat memeriksa apakah 2 lebih besar dari 3.

In [1]:
print(2 > 3)

False


Output dari kode tersebut akan menjadi False. Ini karena pernyataan 2 > 3 mengevaluasi apakah angka 2 lebih besar dari angka 3. Karena 2 tidak lebih besar dari 3, hasilnya adalah False. Jadi, False akan dicetak ke layar sebagai output dari pernyataan print.

In [2]:
var_one = 1
var_two = 2

print(var_one < 1)
print(var_two >= var_one)

False
True


Output dari kode tersebut akan terdiri dari dua baris:

1. Pada baris pertama, print(var_one < 1) mengevaluasi apakah var_one kurang dari 1. Karena var_one adalah 1, dan 1 tidak kurang dari 1, hasilnya adalah False.
2. Pada baris kedua, print(var_two >= var_one) mengevaluasi apakah var_two lebih besar dari atau sama dengan var_one. Karena var_two adalah 2 dan var_one adalah 1, 2 lebih besar dari atau sama dengan 1, sehingga hasilnya adalah True.

## Conditional Statements (Pernyataan Kondisi)

### If Statement (Pernyataan If)

In [4]:
def evaluate_temp(temp):
    # Set an initial message
    message = "Normal temperature."
    # Update value of message only if temperature greater than 38
    if temp > 38:
        message = "Fever!"
    return message

Fungsi evaluate_temp mengambil satu parameter, yaitu temp, yang merupakan suhu yang ingin dievaluasi. Fungsi ini memeriksa apakah suhu melebihi 38 derajat Celsius. Jika ya, itu berarti ada demam dan fungsi akan mengembalikan pesan "Fever!". Jika tidak, pesan "Normal temperature." akan dikembalikan.

In [5]:
print(evaluate_temp(37))

Normal temperature.


Panggilan print(evaluate_temp(37)) akan mengevaluasi suhu 37 derajat Celsius menggunakan fungsi evaluate_temp. Karena 37 tidak lebih besar dari 38, pesan yang dikembalikan akan tetap "Normal temperature.".

In [6]:
print(evaluate_temp(39))

Fever!


Panggilan evaluate_temp(39) akan mengevaluasi suhu 39 derajat Celsius menggunakan fungsi evaluate_temp. Karena 39 lebih besar dari 38, maka pesan yang dikembalikan akan menjadi "Fever!".

### If Else Statement (Pernyataan If Else)

In [7]:
def evaluate_temp_with_else(temp):
    if temp > 38:
        message = "Fever!"
    else:
        message = "Normal temperature."
    return message

Fungsi evaluate_temp_with_else mirip dengan evaluate_temp sebelumnya, tetapi menggunakan blok else. Ini berarti jika kondisi dalam if tidak terpenuhi (yaitu, jika suhu tidak lebih besar dari 38), maka blok else akan dieksekusi.

Jika suhu lebih besar dari 38 derajat Celsius, fungsi ini akan mengembalikan pesan "Fever!". Jika tidak, maka pesan yang dikembalikan akan menjadi "Normal temperature.".

In [8]:
print(evaluate_temp_with_else(37))

Normal temperature.


Panggilan evaluate_temp_with_else(37) akan mengevaluasi suhu 37 derajat Celsius menggunakan fungsi evaluate_temp_with_else. Karena 37 tidak lebih besar dari 38, maka blok else akan dieksekusi, dan pesan yang dikembalikan akan menjadi "Normal temperature.".

### If Elif Else Statement (Pernyataan If Elif Else)

In [9]:
def evaluate_temp_with_elif(temp):
    if temp > 38:
        message = "Fever!"
    elif temp > 35:
        message = "Normal temperature."
    else:
        message = "Low temperature."
    return message

Fungsi evaluate_temp_with_elif juga mengevaluasi suhu, namun kali ini dengan penggunaan pernyataan elif (yang merupakan singkatan dari "else if"). Ini berarti, pertama-tama, kode akan memeriksa apakah suhu melebihi 38 derajat Celsius. Jika ya, itu akan mengembalikan pesan "Fever!".

Jika tidak, maka kode akan memeriksa apakah suhu lebih besar dari 35 derajat Celsius. Jika ya, itu akan mengembalikan pesan "Normal temperature.".

Jika tidak ada kondisi di atas yang terpenuhi, maka blok else akan dieksekusi dan mengembalikan pesan "Low temperature.".

In [10]:
evaluate_temp_with_elif(36)

'Normal temperature.'

Panggilan evaluate_temp_with_elif(36) akan mengevaluasi suhu 36 derajat Celsius menggunakan fungsi evaluate_temp_with_elif.

Karena 36 derajat Celsius tidak memenuhi kondisi pertama if temp > 38, maka kode akan mengevaluasi kondisi kedua elif temp > 35. Karena 36 derajat Celsius memenuhi kondisi tersebut, fungsi akan mengembalikan pesan "Normal temperature.".

In [11]:
evaluate_temp_with_elif(34)

'Low temperature.'

Panggilan evaluate_temp_with_elif(34) akan mengevaluasi suhu 34 derajat Celsius menggunakan fungsi evaluate_temp_with_elif.

Karena 34 derajat Celsius tidak memenuhi kondisi pertama if temp > 38, dan juga tidak memenuhi kondisi kedua elif temp > 35, maka blok else akan dieksekusi. Sehingga, fungsi akan mengembalikan pesan "Low temperature.".

#### Contoh Kalkulasi

In [12]:
def get_taxes(earnings):
    if earnings < 12000:
        tax_owed = .25 * earnings
    else:
        tax_owed = .30 * earnings
    return tax_owed

Fungsi get_taxes mengambil satu parameter, yaitu earnings, yang merupakan penghasilan. Fungsi ini menghitung jumlah pajak yang harus dibayarkan berdasarkan penghasilan yang diberikan.

Jika penghasilan kurang dari $12,000, maka pajak yang harus dibayarkan adalah 25% dari penghasilan (0.25 * earnings). Jika penghasilan sama dengan atau melebihi $12,000, maka pajak yang harus dibayarkan adalah 30% dari penghasilan (0.30 * earnings).

In [13]:
ana_taxes = get_taxes(9000)
bob_taxes = get_taxes(15000)

print(ana_taxes)
print(bob_taxes)

2250.0
4500.0


Output dari kedua pernyataan print() akan bergantung pada penghasilan Ana dan Bob serta logika dalam fungsi get_taxes() yang telah dijelaskan sebelumnya. Pajak yang harus dibayarkan oleh Ana adalah $2,250 (25% dari $9,000), sementara pajak yang harus dibayarkan oleh Bob adalah $4,500 (30% dari $15,000).

#### Contoh Pernyataan Multiple "Elif"

In [14]:
def get_dose(weight):
    # Dosage is 1.25 ml for anyone under 5.2 kg
    if weight < 5.2:
        dose = 1.25
    elif weight < 7.9:
        dose = 2.5
    elif weight < 10.4:
        dose = 3.75
    elif weight < 15.9:
        dose = 5
    elif weight < 21.2:
        dose = 7.5
    # Dosage is 10 ml for anyone 21.2 kg or over
    else:
        dose = 10
    return dose

Fungsi get_dose mengambil satu parameter, yaitu weight, yang merupakan berat pasien. Fungsi ini memberikan dosis obat yang sesuai berdasarkan berat pasien.

Berikut adalah penjelasan dosis obat yang diberikan oleh fungsi ini:

1. Jika berat pasien kurang dari 5.2 kg, dosisnya adalah 1.25 ml.
2. Jika berat pasien kurang dari 7.9 kg (tetapi lebih dari atau sama dengan 5.2 kg), dosisnya adalah 2.5 ml.
3. Jika berat pasien kurang dari 10.4 kg (tetapi lebih dari atau sama dengan 7.9 kg), dosisnya adalah 3.75 ml.
4. Jika berat pasien kurang dari 15.9 kg (tetapi lebih dari atau sama dengan 10.4 kg), dosisnya adalah 5 ml.
5. Jika berat pasien kurang dari 21.2 kg (tetapi lebih dari atau sama dengan 15.9 kg), dosisnya adalah 7.5 ml.
6. Jika berat pasien 21.2 kg atau lebih, dosisnya adalah 10 ml.

In [15]:
print(get_dose(12))

5


Panggilan get_dose(12) akan mengevaluasi dosis obat untuk pasien dengan berat 12 kg menggunakan fungsi get_dose.

Karena berat pasien 12 kg berada di antara 10.4 kg dan 15.9 kg, maka dosis yang diberikan akan menjadi 5 ml sesuai dengan pernyataan elif weight < 15.9.